### PR TITLE
Fix GHA

### PR DESCRIPTION
**Pull request by jharmison-redhat**
This should fix GHA and get us back to building.

### CODE REVIEW

**Comment by jharmison-redhat** [comment]
There are a lot of changes here that are related to updates, some stability things, and some debugging output I found helpful while working on this issue that I think should stay to help us understand CI failures a bit better in the future as well.

**Comment by jharmison-redhat** [comment]
CI is passing but there's an error showing in pip related to requests dependencies being brought in, I'm going to try to pin requests higher to see if it behaves. My local pip is resolving fine, so I'm unsure why Ubuntu is having a hard time with it.

**Comment by jharmison-redhat** [comment]
I could use eyes on this now to see if you think the changes are appropriate, @andykrohg

File: molecule/default/tasks/gitea_test.yml [hunk 1]
```diff
@@ -109,7 +109,10 @@
 
 - name: Verify that Gitea failed to be created from the CR when appropriate
   assert:
-    that: namespace_modded_cr.resources[0].status is not defined == (scope == "namespace")
+    that: namespace_modded_success == namespace_modded_expected
+  vars:
+    namespace_modded_success: '{{ namespace_modded_cr.resources[0].status is defined }}'
+    namespace_modded_expected: '{{ scope == "cluster" }}'
```
Comment (andykrohg) [inline molecule/default/tasks/gitea_test.yml:115]: *Nice.*

Comment (jharmison-redhat) [reply molecule/default/tasks/gitea_test.yml:115]: Yeah it makes it easier to read by a lot, but also improves reliability on newer ansible versions that didn't like the old way, so.....

**Review by andykrohg — APPROVED** [review]

**Pull request merged**